packages/server/billing/helpers/updateSubscriptionQuantity.ts

```diff
@@ -0,0 +1,68 @@
+import getRethink from '../../database/rethinkDriver'
+import {DataLoaderWorker} from '../../graphql/graphql'
+import {getStripeManager} from '../../utils/stripe'
+import insertStripeQuantityMismatchLogging from '../../postgres/queries/insertStripeQuantityMismatchLogging'
+import sendToSentry from '../../utils/sendToSentry'
+import RedisLockQueue from '../../utils/RedisLockQueue'
+
+/**
+ * Check and update if necessary the subscription quantity
+ * @param logMismatch Pass true if a quantity mismatch should be logged
+ */
+const updateSubscriptionQuantity = async (
+  orgId: string,
+  dataLoader: DataLoaderWorker,
+  logMismatch?: boolean
+) => {
+  const r = await getRethink()
+  const manager = getStripeManager()
+
+  const org = await dataLoader.get('organizations').load(orgId)
+  if (!org) throw new Error(`org not found for invoice`)
+  const {stripeSubscriptionId} = org
+  if (!stripeSubscriptionId) return
+
+  const redisLock = new RedisLockQueue(`updateSubscriptionQuantity:${orgId}`, 5000)
+  try {
+    await redisLock.lock(10000)
```

Comment:
-1 i may be misinterpreting this API, but I think this says each lock will last a max of 5 seconds & wait up to 10 seconds to achieve a lock.

if a call to stripe takes 1 second max & we have 2 calls to stripe per function, that means this'll timeout if we have > 5 users joining the org at the same time.
I know in the past we've had something like 50 users migrated to an org. we may want to bump this up by 10-100x just because if it fails this'll be really hard to track down.

Reply:
You're correct with the timeouts for the locks, but I don't think it's an issue. If we have 50 users migrated, we don't need to run this 50 times but actually only once at the end. Because the locked section will read the last user count from the subscription **and** the db, it doesn't matter which of the 50 jobs comes through as long as there is one job which runs after all changes are done. The current setting should achieve this.

I will add it as a comment.